common/src/pathName.C

```diff
@@ -1,326 +1,107 @@
 /*
  * See the dyninst/COPYRIGHT file for copyright information.
- * 
+ *
  * We provide the Paradyn Tools (below described as "Paradyn")
  * on an AS IS basis, and do not warrant its validity or performance.
  * We reserve the right to update, modify, or discontinue this
  * software at any time.  We shall have no obligation to supply such
  * updates or modifications or any other form of support to you.
- * 
+ *
  * By your use of Paradyn, you understand and agree that we (or any
  * other person or entity with proprietary rights in Paradyn) are
  * under no obligation to provide either maintenance services,
  * update services, notices of latent defects, or correction of
  * defects for Paradyn.
- * 
+ *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
  * License as published by the Free Software Foundation; either
  * version 2.1 of the License, or (at your option) any later version.
- * 
+ *
  * This library is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
  * Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
-// pathName.C
-#include "common/src/headers.h"  // P_strrchr()
-#include <ctype.h>
-#include <assert.h>
-#include <limits.h>
-#include "common/src/pathName.h"
-#include <boost/filesystem.hpp>
+#include "pathName.h"
 #include <boost/algorithm/string/replace.hpp>
 #include <boost/algorithm/string/trim.hpp>
-
-#if defined(os_windows) //ccw 20 july 2000 : 29 mar 2001
-	#define S_ISDIR(x) ((x) & _S_IFDIR)
-#else
-	#include <sys/types.h>
-	#include <sys/stat.h>
-	#include <unistd.h>
-	#include <pwd.h>
-#endif
-
-std::string expand_tilde_pathname(const std::string &dir) {
-#ifdef os_windows
```

Comment:
Shouldn't we keep it a no-op for windows?

Reply:
I made a note about that (https://github.com/dyninst/dyninst/pull/1905/commits/141591b6f9199e53655c0ea3caa1674b830e3e5f). These functions are only ever used in linux/freebsd files right now. My goal is to move away from doing OS-specific code by guarding blocks with macros.

Reply:
Now that I've added back the lookup in passwd, I've also needed to add back the windows check. It's at least at the file level instead of inside the functions.